lib/record_store/provider/ns1.rb

```diff
@@ -219,6 +220,35 @@ def symbolize_keys(hash)
       def secrets
         super.fetch('ns1')
       end
+
+      # Patch the method which retrieves headers for API rate limit dynamically
+      module NS1::Transport
+        refine NetHttp do
+          def process_response(response)
+            ratelimit = ['remaining', 'period']
+              .map { |k| [k, response["x-ratelimit-#{k}"].to_i] }.to_h
+
+            ratelimit_remaining = ratelimit["remaining"]
+            ratelimit_period = ratelimit["period"]
+
+            if ratelimit_remaining < 2
```

Comment:
You can probably replace this whole block by:

```
sleep(ratelimit_period / max(1, ratelimit_remaining))
```
Also, not sure about how ruby handles divisions with integers? Might want to convert those to floats?

Reply:
I think just the snippet that you provided is enough. I tested with float type and it slows performance down with when it returns 0.2, 0.3 (integer will be 0, so no waiting time for those).